swift/ql/lib/codeql/swift/dataflow/internal/SsaImplSpecific.qll

```diff
@@ -42,4 +50,12 @@ predicate variableRead(BasicBlock bb, int i, SourceVariable v, boolean certain)
     v = ref.getDecl() and
     certain = true
   )
+  or
+  exists(ReturnStmt return, AbstractFunctionDecl func |
```

Comment:
So at every `ReturnStmt` we implicitly read all of the inout parameters, right? What happens if there's no `ReturnStmt`, but there are still inout parameters that are modified by the function? Does the compiler insert dummy returns for us?

Reply:
We should already have a single exit point for a given function, I think. The `ExitNode` class should represent the `ControlFlowNode` that is the last node of a given callable.

Reply:
Ok, then we can just reuse that.